src/query/interpreter.cpp

```diff
@@ -4025,10 +4093,22 @@ void Interpreter::Commit() {
   // We should document clearly that all results should be pulled to complete
   // a query.
   current_transaction_.reset();
-  if (!current_db_.db_transactional_accessor_) return;
+  if (!current_db_.db_transactional_accessor_ || !current_db_.db_acc_) {
+    // No database nor db transaction; check for system transaction
+    if (!system_transaction) return;
+    // TODO auto expected = TransactionStatus::SYSTEM;
```

Comment:
Do we need to remove the SYSTEM enum?

Reply:
I would like to have a way to distinguish between system and data transactions.
Removed for now.